File: fopa_sop_apk/lib/features/feat_category/datas/dtos/create_category_dto.dart
class CreateCategoryDto {
  final String name;
  final String? description;

  CreateCategoryDto({required this.name, this.description});

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{'name': name};
    if (description != null) map['description'] = description;
    return map;
  }
}

class UpdateCategoryDto {
  final String name;
  final String? description;

  UpdateCategoryDto({required this.name, this.description});

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{'name': name};
    if (description != null) map['description'] = description;
    return map;
  }
}

File: src/application/dtos/category/create-category.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional, IsString } from 'class-validator';

export class CreateCategoryDto {
  @ApiProperty({ example: 'Eaux' })
  @IsNotEmpty({ message: 'Le nom de la catégorie est requis' })
  @IsString()
  name: string;

  @ApiProperty({ example: 'Eaux minérales et gazeuses', required: false })
  @IsOptional()
  @IsString()
  description?: string;
}
